gateway/1.8/modules/ROOT/pages/flex-security-best-practices.adoc

```diff
@@ -0,0 +1,164 @@
+
+Certificates exposed to the internet are never used internally and certificates that are used internally are never exposed to the internet. Separating internal and external certificates prevents potential leakage of the internal credentials.
+
+You can use Flex Gateway as at both the runtime environment ingress layer or at the inner environment layer. For Flex Gateways at the ingress layer, limit the gateways scope to TLS termination, routing, and other basic protections. Push more CPU and memory intensive operations to inner Flex Gateways.
+
+[[protect-apis-by-using-policies]]
+=== Protect APIs by using Policies
+
+To further configure API protections, apply policies. Flex Gateway offers xref:policies-included-directory.adoc[Included Policies] and enables you to develop custom policies by using xref:pdk::policies-pdk-overview.adoc[].
+
+[[govern-apis-by-using-api-governance]]
+=== Govern APIs by using API Governance
+
+Use xref:api-governance::index.adoc[API Governance] to ensure APIs are designed and deployed in line with MuleSoft, industry, and your organization's best practices. 
+
+[[protect-the-upstream-service]]
+== Protect the Upstream Service (Flex Gateway to API Provider)
+
+Protecting the flow of traffic from Flex to the upstream service is dependent on the location of the upstream service.
+
+[[protect-services-available-locally-to-flex-gateway]]
+=== Protect Services Available Locally to Flex Gateway
+
+For services in the same network as Flex Gateway, security restrictions might be lower.
+
+The organization must define whether plain or TLS connections are in place. If using TLS, configure mTLS via Flex Gateway Outbound TLS contexts. To learn more about configuring TLS, see xref:
+
+* xref:flex-conn-tls-config.adoc[]
+* xref:flex-local-tls-config.adoc[]
+
+Additionally, configure the upstream service to only accept traffic incoming from Flex Gateway. This protects the upstream service from other services in the same network. For Kubernetes, configure this using Network Policies. 
+
+NOTE: To find reference architecture detailing communication to an internal upstream service, see xref:flex-architecture-uc1.adoc[] and xref:flex-architecture-uc3.adoc[].
+
+[[protect-external-services]]
+=== Protect External Services
+
+When protecting external services not in the same network, connect to the service using mTLS communication. Additionally, configure a trusted network link (private VPC link or other) between the networks to enable Flex Gateway to securely access the upstream service without exposing the service to the public internet. 
+
+NOTE: To find reference architecture detailing communication to an external upstream service, see xref:flex-architecture-uc2.adoc[] and xref:flex-architecture-uc4.adoc[].
+
+[[preform-regular-testing]]
+== Preform Regular Testing
+
+Perform regular penetration and vulnerability tests to identify new or regressed issues in your Flex Gateway deployment and exposed APIs. When testing, consider:
```

Comment:
```suggestion
Perform regular penetration and vulnerability tests to identify new or regressed issues in your Flex Gateway deployment and exposed APIs. Consider these when testing:
```

Reply:
I think a word needs to be added between "these when" in the second sentence, but I wasn't sure what to suggest. Maybe "points" or "configurations"?